Game/ConsoleApplication2/MovingGameObject.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "GameObject.h"
+#include "Constants.h"
+
+
+class MovingGameObject : public GameObject
+{
```

Comment:
Almost every game object can move (i.e update their position and speed). I've currently got Player, Bullet and Astronaut doing this using common functions. I would think it'd be unnecessary to duplicate that code. It also fits the 'is-a' inheritance model as well. I'd personally think it'd be odd not to adopt it.

As for AIObject, what common functionality could we add to that class? AI Objects just call a static method from their update, and I don't think any specific behavior is duplicated.